cpp/src/Ice/ThreadPool.cpp

```diff
@@ -199,6 +207,22 @@ IceInternal::ThreadPool::ThreadPool(const InstancePtr& instance, const string& p
 #endif
       _promote(true)
 {
+#ifdef __APPLE__
+    if (_dispatchQueue && _executor)
+    {
+        throw InitializationException{__FILE__, __LINE__, "cannot use both dispatch queues and custom executors"};
+    }
+
+    if (_dispatchQueue)
+    {
+        _executor = [&](function<void()> call, const Ice::ConnectionPtr&)
```

Comment:
I think `[this]` would be clearer than `[&]`. Or maybe you can capture only the dispatch queue by copy (I assume it's a pointer?): `[dispatchQueue = _dispatchQueue]`?